frontend/apps/reader/modules/readerlink.lua

```diff
@@ -91,6 +92,138 @@ function ReaderLink:init()
+        if wiki_lang and wiki_page then
+            -- Ask for user confirmation before launching lookup (on a
+            -- wikipedia page saved as epub, full of wikipedia links, it's
+            -- too easy to click on links when wanting to change page...)
+            -- But first check if this wikipedia article has been saved as EPUB
+            local epub_filename = wiki_page .. "."..string.upper(wiki_lang)..".epub"
+            local epub_fullpath
+            -- either in current book directory
+            local last_file = G_reader_settings:readSetting("lastfile")
+            if last_file then
+                local current_book_dir = last_file:match("(.*)/")
+                local safe_filename = util.getSafeFilename(epub_filename, current_book_dir):gsub("_", " ")
+                local epub_path = current_book_dir .. "/" .. safe_filename
+                if util.pathExists(epub_path) then
+                    epub_fullpath = epub_path
+                end
+            end
+            -- or in wikipedia save directory
+            if not epub_fullpath then
+                local dir = G_reader_settings:readSetting("wikipedia_save_dir")
+                if not dir then dir = G_reader_settings:readSetting("home_dir") end
+                if not dir then dir = require("apps/filemanager/filemanagerutil").getDefaultDir() end
+                if dir then
+                    local safe_filename = util.getSafeFilename(epub_filename, dir):gsub("_", " ")
+                    local epub_path = dir .. "/" .. safe_filename
+                    if util.pathExists(epub_path) then
+                        epub_fullpath = epub_path
+                    end
+                end
+            end
+            return epub_fullpath
+        else
+            return false
+        end
+    end
+    self._external_buttons["06_wiki_saved"] = function(this, link_url)
+        return {
+            text = _("Read EPUB"),
+            callback = function()
+                UIManager:scheduleIn(0.1, function()
+                        UIManager:close(this.dialog)
+                        self.ui:switchDocument(find_epub_fullpath(link_url))
+                end)
+            end,
+            show_in_dialog_func = function()
+                local wiki_lang, wiki_page = link_url:match([[https?://([^%.]+).wikipedia.org/wiki/([^/]+)]])
+                if wiki_lang and wiki_page then
+                    if find_epub_fullpath(link_url) then
+                        local text = T(_("This article has previously been saved as EPUB. You may wish to read the saved EPUB instead."))
+                        return true, text
+                    else
+                        return false
+                    end
+                else
+                    return false
+                end
+            end
+        }
+    end
+    self._external_buttons["07_cancel"] = function(this, link_url)
```

Comment:
Where would you place your future buttons?
We had some discussions with the highlight dialog buttons, as depending on stuff, we may get an even or an odd number of buttons, and so the last one may be wide (when even number). We decided for "Search" to be that last.
For external links, "Cancel" is obviously the candidate, so no real problem there :)

But may be number the names better ? 10_copy ...  90_cancel so your plugin can more easily insert itself between these ?